### PR TITLE
Add convenience catalog function to list all available graphs

### PR DESCRIPTION
**Pull request by Mats-SX**

### CODE REVIEW

File: okapi-api/src/main/scala/org/opencypher/okapi/api/graph/CypherSession.scala [hunk 1]
```diff
@@ -41,22 +38,23 @@ import org.opencypher.okapi.impl.io.SessionGraphDataSource.{Namespace => Session
 trait CypherSession {
 
   /**
-    * The [[org.opencypher.okapi.api.graph.Namespace]] used to to store graphs within this session.
+    * Executes a Cypher query in this session on the current ambient graph.
     *
-    * @return session namespace
+    * @param query      Cypher query to execute
+    * @param parameters parameters used by the Cypher query
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/api/graph/CypherSession.scala]: We are missing a parameter doc for `drivingTable`

File: okapi-api/src/main/scala/org/opencypher/okapi/api/graph/CypherSession.scala [hunk 2]
```diff
@@ -41,22 +38,23 @@ import org.opencypher.okapi.impl.io.SessionGraphDataSource.{Namespace => Session
...
+    * @return result of the query
     */
-  def sessionNamespace: Namespace
+  def cypher(query: String, parameters: CypherMap = CypherMap.empty, drivingTable: Option[CypherRecords] = None): CypherResult
 
   /**
-    * Stores a mutable mapping between a data source [[org.opencypher.okapi.api.graph.Namespace]] and the specific [[org.opencypher.okapi.api.io.PropertyGraphDataSource]].
+    * Access this sessions [[org.opencypher.okapi.api.graph.PropertyGraphCatalog]].
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/api/graph/CypherSession.scala]: Maybe align a bit more with `SparkSession.catalog`:
```
Interface through which the user may (de-)register property graph datasources as well as read, write and delete property graphs.
```

File: okapi-api/src/main/scala/org/opencypher/okapi/api/graph/CypherSession.scala [hunk 3]
```diff
@@ -41,22 +38,23 @@ import org.opencypher.okapi.impl.io.SessionGraphDataSource.{Namespace => Session
...
-    * This mapping also holds the [[org.opencypher.okapi.impl.io.SessionGraphDataSource]] by default.
+    * @return session catalog
     */
-  protected var dataSourceMapping: Map[Namespace, PropertyGraphDataSource] =
-    Map(sessionNamespace -> new SessionGraphDataSource)
+  def catalog: PropertyGraphCatalog
 
   /**
-    * Register the given [[org.opencypher.okapi.api.io.PropertyGraphDataSource]] under the specific [[org.opencypher.okapi.api.graph.Namespace]] within this session.
+    * Register the given [[org.opencypher.okapi.api.io.PropertyGraphDataSource]] under the specific [[org.opencypher.okapi.api.graph.Namespace]] within the sessions catalog.
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/api/graph/CypherSession.scala]: maybe just call it `the session catalog`

File: okapi-api/src/main/scala/org/opencypher/okapi/api/graph/CypherSession.scala [hunk 4]
```diff
@@ -66,112 +64,16 @@ trait CypherSession {
     * @param namespace  namespace for lookup
     * @param dataSource property graph data source
     */
-  def registerSource(namespace: Namespace, dataSource: PropertyGraphDataSource): Unit = dataSourceMapping.get(namespace) match {
-    case Some(p) => throw IllegalArgumentException(s"no data source registered with namespace '$namespace'", p)
-    case None => dataSourceMapping = dataSourceMapping.updated(namespace, dataSource)
-  }
+  def registerSource(namespace: Namespace, dataSource: PropertyGraphDataSource): Unit =
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/api/graph/CypherSession.scala]: I was wondering if we should still have these mirrored methods on `CypherSession`. In a program `caps.catalog.registerSource(...)` looks convenient (and spark-aligned) enough for me. Maybe @alastai has an opinion on that?

File: okapi-api/src/main/scala/org/opencypher/okapi/api/graph/PropertyGraphCatalog.scala [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+  /**
+    * Returns all [[org.opencypher.okapi.api.graph.Namespace]]s registered at this catalog.
+    *
+    * @return registered namespaces
+    */
+  def namespaces: Set[Namespace]
+
+  /**
+    * Returns all available [[org.opencypher.okapi.api.io.PropertyGraphDataSource]]s.
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/api/graph/PropertyGraphCatalog.scala]: available -> registered

File: okapi-api/src/main/scala/org/opencypher/okapi/api/graph/PropertyGraphCatalog.scala [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    *
+    * @return a map of all PGDS known to this catalog, keyed by their [[org.opencypher.okapi.api.graph.Namespace]]s.
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/api/graph/PropertyGraphCatalog.scala]: known to -> registered at

File: okapi-api/src/main/scala/org/opencypher/okapi/impl/graph/CypherCatalog.scala [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+import org.opencypher.okapi.impl.exception.{IllegalArgumentException, UnsupportedOperationException}
+import org.opencypher.okapi.impl.io.SessionGraphDataSource
+
+/**
+  * This is the default implementation of the [[org.opencypher.okapi.api.graph.PropertyGraphCatalog]].
+  * It uses a mutable mapping to store the mapping between
+  * [[org.opencypher.okapi.api.graph.Namespace]]s and [[org.opencypher.okapi.api.io.PropertyGraphDataSource]]s.
+  *
+  * By default this catalog mounts a single [[org.opencypher.okapi.impl.io.SessionGraphDataSource]] under the namespace
+  * [[org.opencypher.okapi.impl.graph.CypherCatalog#sessionNamespace]]. This is PGDS is used to session local graphs.
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/impl/graph/CypherCatalog.scala]: to -> to store | for

File: okapi-api/src/main/scala/org/opencypher/okapi/impl/graph/CypherCatalog.scala [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+  */
+class CypherCatalog extends PropertyGraphCatalog{
+
+  /**
+    * The [[org.opencypher.okapi.api.graph.Namespace]] used to to store graphs within this session.
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/impl/graph/CypherCatalog.scala]: to to -> to

File: okapi-api/src/main/scala/org/opencypher/okapi/impl/graph/CypherCatalog.scala [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    *
+    * @return session namespace
+    */
+  def sessionNamespace: Namespace = SessionGraphDataSource.Namespace
+
+  /**
+    * Stores a mutable mapping between a data source [[org.opencypher.okapi.api.graph.Namespace]] and the specific
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/impl/graph/CypherCatalog.scala]: between a ~~data source~~

File: okapi-api/src/main/scala/org/opencypher/okapi/api/graph/PropertyGraphCatalog.scala [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+  def listSources: Map[Namespace, PropertyGraphDataSource]
+
+  /**
+    * Returns the [[org.opencypher.okapi.api.io.PropertyGraphDataSource]] that is registered under
+    * the given [[org.opencypher.okapi.api.graph.Namespace]].
+    *
+    * @param namespace namespace for lookup
+    * @return property graph data source
+    */
+  def source(namespace: Namespace): PropertyGraphDataSource
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/api/graph/PropertyGraphCatalog.scala]: maybe call it `dataSource`?

Comment (DarthMax) [reply okapi-api/src/main/scala/org/opencypher/okapi/api/graph/PropertyGraphCatalog.scala]: I was not sure here. We call it `registerSource` so either we change it to `registerDataSource` or we keep `source`

Comment (s1ck) [reply okapi-api/src/main/scala/org/opencypher/okapi/api/graph/PropertyGraphCatalog.scala]: I'm fine with both, as long as it's consistent. Whatever you prefer.

**Pull request merged**